File: test/bench/md5/go/md5-bench.go
// md5-bench — Go MD5 基準測試（標準庫）
//
// 測量 100,000 次 MD5（"abc"，每次修改 input[2] 防止死代碼消除）
//
// 編譯：go build -ldflags="-s -w" -o md5-bench_go .
// 執行：./md5-bench_go

package main

import (
	"crypto/md5"
	"fmt"
)

func main() {
	input := []byte("abc")
	var result [16]byte

	for i := 0; i < 100000; i++ {
		input[2] = byte(i & 255)
		result = md5.Sum(input)
	}

	// 輸出以防止死代碼消除
	fmt.Println(result[0])
}

File: test/bench/md5/c/md5-bench.c
// md5-bench — C MD5 benchmark (macOS CommonCrypto)
//
// 100,000 iterations of MD5("abc", modifying input[2] each iteration)
//
// Build: clang -O3 -o md5-bench_c md5-bench.c
// Run:   ./md5-bench_c

#include <stdio.h>
#include <CommonCrypto/CommonDigest.h>

int main() {
    unsigned char input[3] = {'a', 'b', 'c'};
    unsigned char result[CC_MD5_DIGEST_LENGTH];

    for (int i = 0; i < 100000; i++) {
        input[2] = (unsigned char)(i & 255);
        CC_MD5(input, 3, result);
    }

    // Output to prevent dead code elimination
    printf("%d\n", result[0]);
    return 0;
}
